src/api/java/com/minecolonies/api/util/ItemStackUtils.java

```diff
@@ -991,5 +988,41 @@ public static ItemStack idToItemStack(final String itemData)
         }
         return stack;
     }
+
+    /**
+     * Obtains a list of all basic items in the game, plus any extra items present in the player's
+     * inventory (allowing for items with custom NBT, e.g. DO blocks or dyed armour).
+     *
+     * @param player The player whose inventory to check.
+     * @return The set of items.
+     */
+    public static Set<ItemStack> allItemsPlusInventory(@NotNull final PlayerEntity player)
+    {
+        // using ItemStorage for better hashing
+        final Set<ItemStorage> allItems = new HashSet<>();
+
+        // all known items
+        for (final ItemStack stack : IColonyManager.getInstance().getCompatibilityManager().getListOfAllItems())
+        {
+            allItems.add(new ItemStorage(stack, true));
+        }
+
+        // plus all items from the player's inventory not already listed (adds items with extra NBT)
+        for (final ItemStack stack : player.inventory.items)
+        {
+            if (stack.isEmpty()) continue;
+
+            ItemStack pristine = stack;
+            if (stack.isDamageableItem() && stack.isDamaged())
+            {
+                pristine = stack.copy();
+                stack.setDamageValue(0);
+                // in case the item wasn't already in the set, we want to only store a pristine one!
```

Comment:
Shouldn’t this be the `pristine` ItemStack? Now the item stack in the players inventory gets modified, and the damaged one gets stored in `allItems`